### ISSUE TITLE
lint_single_char_pattern shows extra quotation mark and number sign with raw string literal

### ISSUE THREAD

**Issue by IgaguriMK** [rust-lang/rust-clippy#4356]
## Problem

With the following code,

```rust
fn main() {
    for s in "try using a char instead".split(r"a") {
        println!("{}", s);
    }

    for s in "try using a char instead".split(r#"a"#) {
        println!("{}", s);
    }
}
```

clippy shows invalid char literals: `'"a'` and `'#"a"'` .

```
warning: single-character string constant used as pattern
 --> src\main.rs:2:47
  |
2 |     for s in "try using a char instead".split(r"a") {
  |                                               ^^^^ help: try using a char instead: `'"a'`
  |
  = note: #[warn(clippy::single_char_pattern)] on by default
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#single_char_pattern

warning: single-character string constant used as pattern
 --> src\main.rs:6:47
  |
6 |     for s in "try using a char instead".split(r#"a"#) {
  |                                               ^^^^^^ help: try using a char instead: `'#"a"'`
  |
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#single_char_pattern
```

## Clippy version
```
clippy 0.0.212 (082cfa79 2019-06-07)
```

**Comment by phansch** [comment]
Thanks for the report! On first glance, it looks like the lint suggestion currently doesn't handle raw strings at all:

https://github.com/rust-lang/rust-clippy/blob/b041511b5fcd386c4ae74a30b60a5081f8717fbe/clippy_lints/src/methods/mod.rs#L2337-L2338
